### ISSUE TITLE
deleteDir removes other files/dirs

### ISSUE THREAD

**Issue by jeroenlammerts** [chrisnharvey/flysystem-openstack-swift#19]
For example the following folder structure:

```
mydir/subdir
mydir/subdir2

```
If you delete "mydir/subdir" it is removing all files under "mydir/subdir2" too.

```
$objects = $this->container->listObjects([
    'prefix' => $this->applyPathPrefix($dirname)
]);
```

This part is returning files matching like "mydir/subdir*". I think the dirname always should contain a trailing slash.

**Comment by chrisnharvey** [comment]
You’re absolutely right.

I’ll get this fixed soon and tag a new release.

**Comment by peter279k** [comment]
@chrisnharvey, do you have any plan to work on this?

Please describe the approach and I can implement your approach for a PR :).

**Comment by mzur** [comment]
I think this suggestion should work:

> I think the dirname always should contain a trailing slash.

So you check if the provided dirname has a trailing slash and append one if it doesn't.

**Comment by chrisnharvey** [comment]
Fixed in v0.3.4

Thanks @jeroenlammerts
